### PR TITLE
Add Install from tar guide

### PR DESCRIPTION
**Pull request by BenLloydPearson**
#### Summary
We are making changes to how we present information on our [deploy](mattermost.com/deploy) and [download](mattermost.com/download) web pages and we need a documentation page that provides non-platform specific information on how to install Mattermost from the tarball releases we make. This content is almost a direct replica of the installation instructions for [Debian Buster](https://docs.mattermost.com/install/install-debian.html#installing-mattermost-server) with some slight tweaks to the intro/outro.

I have two unanswered questions:
1. Is installing from tarball synonymous with installing from source? I like the latter language better but I want to make sure its technically correct.
2. Since this is a replica of other content, should we replace the Debian instructions and reference a single file in both locations?

#### Ticket Link
n/a

### CODE REVIEW

**Comment by cwarnermm** [comment]
@nab-77 Can help us with the terminology question. While I welcome opportunities to include the term "tarball" in our product docs ;) (because who wouldn't really?!?), if "source" is technically equivalent, I agree that it's the term that will speak to the widest audience. The term "tarball" will only be understood by a subset of potential audiences. 

We should absolutely be writing content once and reusing it where appropriate. Docs can help with a content reuse strategy using ``include`` directives in our RST files.

**Comment by BenLloydPearson** [comment]
After speaking with @nab-77 , we decided to change this to Install from Tar and I replaced the Debian instructions according to the request from @cwarnermm. The Debian Buster page is the only place that references the doc page that this PR deprecates.

File: source/install/install-tar.rst [hunk 1]
```diff
@@ -139,4 +134,4 @@ Assume that the IP address of this server is 10.10.10.2.
 
     ``sudo systemctl enable mattermost.service``
 
-Now that the Mattermost server is up and running, you can do some initial configuration and setup.
+Once you're Mattermost server is up and running, take a look at the `available configuration options <https://docs.mattermost.com/guides/deployment.html#deployment-guide./>`__
```
Comment (cwarnermm) [inline source/install/install-tar.rst:137]:
```suggestion
Once you're Mattermost server is up and running, take a look at the `available configuration options <https://docs.mattermost.com/guides/deployment.html#deployment-guide./>`__.
```

File: source/guides/deployment.rst [hunk 2]
```diff
@@ -21,6 +21,7 @@ These guides will get you up and running with Mattermost in minutes.
 
 * :doc:`Quick install guide </getting-started/light-install>` - Deploy in minutes via Mattermost Omnibus on Ubuntu.
 * :doc:`Run Mattermost via Docker </install/setting-up-local-machine-using-docker>` - Launch a Mattermost server instantly to test functionality and build integrations.
+* :doc:`Install Mattermost from Tar </install/install-tar>` - Install Mattermost server from a compressed tarball.
```
Comment (cwarnermm) [inline source/guides/deployment.rst:24]: You'll need to add this new page to the toctree above to surface it up via the LHS navigation pane.

**Review by nab-77 — APPROVED** [review]
Super clear update. I also assume @BenLloydPearson we will pull this content into the relevant tab on the reworked /deploy page.

**Comment by BenLloydPearson** [comment]
> Super clear update. I also assume @BenLloydPearson we will pull this content into the relevant tab on the reworked /deploy page.

@nab-77 we decided not to go the embed direction because that introduces a risk of having docs changes ripple into breaking changes on our website. Instead, we will direct people who download the tarball to visit this docs page.

**Comment by BenLloydPearson** [comment]
> @BenLloydPearson - Great update! This new page isn't yet accessible from the navigation pane. You'll need to add this new page to the toctree on the landing page.

@cwarnermm This should be all set now.

**Comment by nab-77** [comment]
> > Super clear update. I also assume @BenLloydPearson we will pull this content into the relevant tab on the reworked /deploy page.
> 
> @nab-77 we decided not to go the embed direction because that introduces a risk of having docs changes ripple into breaking changes on our website. Instead, we will direct people who download the tarball to visit this docs page.

There are so many great examples wtih ISVs where you do layout steps on such a /deploy (or /install) page. Why would we not include the tarball install steps (as we are doing with Docker) on this page and save that click out to docs? This page is all about getting an instance up and running.

File: source/guides/deployment.rst [hunk 3]
```diff
@@ -39,8 +40,9 @@ Server installation
     :hidden:
 
     Software and hardware requirements </install/software-hardware-requirements>
-    Local Docker setup </install/setting-up-local-machine-using-docker>
+    Installing Mattermost via Docker </install/setting-up-local-machine-using-docker>
```
Comment (cwarnermm) [inline source/guides/deployment.rst:43]:
```suggestion
    Install Mattermost via Docker </install/setting-up-local-machine-using-docker>
```

File: source/guides/deployment.rst [hunk 4]
```diff
@@ -10,7 +10,7 @@ Get started
     :hidden:
 
     Quick install guide </getting-started/light-install>
-    Run Mattermost via Docker </install/setting-up-local-machine-using-docker>
+    Installing Mattermost via Docker </install/setting-up-local-machine-using-docker>
```
Comment (cwarnermm) [inline source/guides/deployment.rst:13]:
```suggestion
    Install Mattermost via Docker </install/setting-up-local-machine-using-docker>
```

**Pull request merged**